qutip/core/gates.py

```diff
@@ -72,10 +83,11 @@ def cz_gate(*, dtype="csr"):
     result : :class:`.Qobj`
         Quantum object for operator describing the rotation.
     """
-    return qdiags([1, 1, 1, -1], dims=[[2, 2], [2, 2]], dtype=dtype)
+    dtype = dtype or settings.core["default_dtype"] or _data.CSR
+    return qdiags([1, 1, 1, -1], dims=_DIMS_2_QB, dtype=dtype)
```

Comment:
How does qdiags handle the isherms property? Will it check if the input is real? If so, maybe for those we could manually set the isherms and isunitary.

Reply:
When there is only one diagonal, it sets it automatically. 
There is only one case in our code base where this is not the case (and it's not in the tests...).
Improving tests for it and adding manual flag setting could make a good first issue.